### PR TITLE
fix(date-textbox): add guard against document for server side

### PR DESCRIPTION
**Pull request by LuLaValva**
## Description

Added a type guard against document for date textbox during server rendering.

## Context

The document object is not accessible from the server side.

### CODE REVIEW

**Review by agliga — APPROVED** [review]

File: src/components/ebay-date-textbox/component.js [hunk 1]
```diff
@@ -66,10 +66,12 @@ export default class extends Marko.Component {
     }
 
     calculateNumMonths() {
-        this.state.numMonths =
-            document.documentElement.clientWidth < MIN_WIDTH_FOR_DOUBLE_PANE
-                ? 1
-                : 2;
+        if (document !== undefined) {
```
Comment (PrashantAshok) [inline src/components/ebay-date-textbox/component.js:69]: This check won't work, `document` would be undefined in server side. You might have to change this to
```
typeof document === 'undefined'
```

Comment (ArtBlue) [reply src/components/ebay-date-textbox/component.js:69]: Can't we just check whether `document` is falsy? And to keep conditionals shorter and the code easier to read, it's usually better to exit functions as quickly as possible...

`````js
if (!document) {
  return;
}
...
`````

Comment (agliga) [reply src/components/ebay-date-textbox/component.js:69]: Yes we need to check for typeof. This can throw a compile error if document is undefined.

**Review by agliga — CHANGES_REQUESTED** [review]
Need to change to check typeof

File: src/components/ebay-date-textbox/component.js (hunk 1 above)
Comment (LuLaValva) [reply src/components/ebay-date-textbox/component.js]: Thanks for the input everyone, I've updated it now

File: src/components/ebay-date-textbox/component.js [hunk 2]
```diff
@@ -66,10 +66,12 @@ export default class extends Marko.Component {
     }
 
     calculateNumMonths() {
-        this.state.numMonths =
-            document.documentElement.clientWidth < MIN_WIDTH_FOR_DOUBLE_PANE
-                ? 1
-                : 2;
+        if (typeof document !== "undefined") {
```
[the remark this answers is not in the document]
Comment (agliga) [reply src/components/ebay-date-textbox/component.js:69]: Agreed. However I'm fine with it as is for now.

**Review by agliga — APPROVED** [review]
Approving. We can change it to a guard if we like. Either way is fine by me.

**Review by agliga — APPROVED** [review]
Approving. We can change it to a guard if we like. Either way is fine by me.

**Review by agliga — APPROVED** [review]

**Pull request merged**